data/cell.hh

```diff
@@ -181,7 +181,7 @@ struct cell {
                 && std::is_same_v<Allocator, imr::alloc::object_allocator::sizer>)
             || (imr::is_serializer_for_v<variable_value_structure, Serializer>
                 && std::is_same_v<Allocator, imr::alloc::object_allocator::serializer>)
-        auto operator()(Serializer serializer, Allocator allocations) {
+        imr::sizing_or_serializing_result auto operator()(Serializer serializer, Allocator allocations) {
```

Comment:
If serializer has a continuation, the return value will be another serializer, which continues where the caller left off. See an example in `imr::object::do_make()` (in `imr/utils.hh`). The inner struct's serializer is passed a serializer containing a continuation with the outer struct's serialization progress. When the innter struct's serializer returns, the serialization of the outer struct is continued.

Maybe this particular writer is not used in this way.

Reply:
Well, it compiled. That doesn't mean it's correct.

Shall I drop the patch?

Reply:
Yes, it happens to work now, but the return type depends on the type of Serializer (whether it has any continuations).